src/PLLModelGetLanguagesListDynamicMethodReturnTypeExtension.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+/**
+ * Set return type of PLL_Model->get_languages_list().
+ */
+
+declare(strict_types=1);
+
+namespace WPSyntex\Polylang\PHPStan;
+
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\PropertyFetch;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Type\Constant\ConstantArrayType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\ArrayType;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\StringType;
+use PHPStan\Type\Type;
+use PHPStan\Type\UnionType;
+use PLL_Language;
+use PLL_Model;
+
+class PLLModelGetLanguagesListDynamicMethodReturnTypeExtension implements DynamicMethodReturnTypeExtension {
+	public function getClass(): string {
+		return PLL_Model::class;
+	}
+
+	public function isMethodSupported( MethodReflection $methodReflection ): bool {
+		return $methodReflection->getName() === 'get_languages_list';
+	}
+
+	public function getTypeFromMethodCall( MethodReflection $methodReflection, MethodCall $methodCall, Scope $scope ): Type {
+		$args = $methodCall->getArgs();
+
+		// Called without arguments.
+		if ( count( $args ) === 0 ) {
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+		}
+
+		$argumentType = $scope->getType( $args[0]->value );
+
+		// Called with an argument that is not an array.
+		if ( ! $argumentType instanceof ConstantArrayType ) {
+			return new ArrayType( new IntegerType(), new ObjectType( PLL_Language::class ) );
+		}
+
+		foreach ( $argumentType->getKeyTypes() as $index => $key ) {
```

Comment:
`getKeyTypes()` is available only for `ConstantArrayType`. Maybe we should retrieve key values for `ArrayType` too, with something such as:
```php
if ($argumentType instanceof ArrayType) {
 	$argsKeys   = $argsType->getKeysArray();
 	$argsValues = $argsType->getValuesArray();
 }
 ```

Reply:
I made some changes with [228d36d](https://github.com/polylang/polylang-phpstan/pull/3/commits/228d36db5ad82939ac2cb5d1da55bfa10be8c3e2) according to your suggestions.